starter_code/public/javascript/index.js

```diff
@@ -1,23 +1,80 @@
-const charactersAPI = new APIHandler("http://localhost:8000")
+const charactersAPI = new APIHandler("http://localhost:8000");
 
-$(document).ready( () => {
-  document.getElementById('fetch-all').onclick = function(){
+window.onload = () => {
+  document.getElementById("fetch-all").onclick = function() {
+    charactersAPI
+      .getFullList()
+      .then(res => {
+        console.log(res);
+        $(".characters-container").empty();
+        res.forEach(element => {
+          $(".characters-container").append(
+            `<div class="character-info">
+            <div class="Id">Id: ${element.id}</div>
+          <div class="name">Name: ${element.name}</div>
+          <div class="occupation">Occupation: ${element.occupation}</div>
+          <div class="debt">Debt: ${element.debt}</div>
+          <div class="weapon">Weapon: ${element.weapon}</div> 
+          </div>`
+          );
+        });
+      })
+      .catch();
+  };
 
-  }
-  
-  document.getElementById('fetch-one').onclick = function(){
-    
-  }
-  
-  document.getElementById('delete-one').onclick = function(){
-        
-  }
-  
-  document.getElementById('edit-character-form').onsubmit = function(){
-            
-  }
-  
-  document.getElementById('new-character-form').onsubmit = function(){
-                
-  }
-})
+  document.getElementById("fetch-one").onclick = function() {
+    const characterId = $(".charID").val();
+    console.log(characterId);
+    charactersAPI
+      .getOneRegister(characterId)
+      .then(res => {
+        $(".characters-container").empty();
+        $(".characters-container").append(
+          `<div class="character-info">
+          <div class="Id">Id: ${res.id}</div>
+        <div class="name">Name: ${res.name}</div>
+        <div class="occupation">Occupation: ${res.occupation}</div>
+        <div class="debt">Debt: ${res.debt}</div>
+        <div class="weapon">Weapon: ${res.weapon}</div> 
+        </div>`
+        );
+      })
+      .catch();
+  };
+
+  document.getElementById("delete-one").onclick = function() {
+    const characterId = $(".charDelete").val();
+    console.log(characterId);
+    charactersAPI
+      .deleteOneRegister(characterId)
+      .then($("#delete-one").css("background-color", "#208c3d"))
+      .catch($("#delete-one").css("background-color", "red"));
+  };
+
+  document.getElementById("edit-character-form").onsubmit = function() {
+    const characterId = $(".editChar").val();
+    const editCharacter = {
+      name: $(".editName").val(),
```

Comment:
Be careful with selecting elements like this. When you use a class selector, you get a collection of elements, not the element itself. This might not work